ebos/eclproblem.hh

```diff
@@ -735,12 +735,15 @@ public:
     void writeOutput(bool verbose = true)
     {
         OPM_TIMEBLOCK(problemWriteOutput);
+
+        bool isSubStep = !EWOMS_GET_PARAM(TypeTag, bool, EnableWriteAllSolutions) && !this->simulator().episodeWillBeOver();
```

Comment:
I think we need to used a different variable for the decision.

Changing isSubStep_ here will also influence how results are written with Damaris and for Eclipse. See the lines below:
```
damarisWriter_->writeOutput(localCellData, isSubStep) ;
eclWriter_->writeOutput(std::move(localCellData), isSubStep);
``

Reply:
afaict there is no change to the variable, it is only moved further up right? i think the name is highly misleading however.

Reply:
> afaict there is no change to the variable, it is only moved further up right? i think the name is highly misleading however.

That is right, there is no change to the `isSubStep` variable from the original code, it is only moved before the VTK output call.  We could change to:
```
if (EWOMS_GET_PARAM(TypeTag, bool, EnableWriteAllSolutions) || this->simulator().episodeWillBeOver()){
            ParentType::writeOutput(verbose);
        }
```
if this is preferred.

Reply:
Sorry, then it was me being dense. No need to change.

Reply:
I updated the code to the above form (and right after I saw your response @blattms, sorry for that).

Reply:
No worries and  thanks.